src/V3Const.cpp

```diff
@@ -2205,10 +2226,23 @@ class ConstVisitor final : public VNVisitor {
             AstNodeExpr* srcp = nodep->rhsp()->unlinkFrBack();
             const int sWidth = srcp->width();
             const int dWidth = dstp->width();
-            if (dWidth == 0) {
-                srcp = new AstCvtPackedToDynArray{nodep->fileline(), srcp, dstp->dtypep()};
-            } else if (sWidth > dWidth) {
-                srcp = new AstSel{streamp->fileline(), srcp, sWidth - dWidth, dWidth};
+            if (VN_IS(dstp->dtypep(), UnpackArrayDType)) {
+                const int dstBitWidth
+                    = dWidth * VN_AS(dstp->dtypep(), UnpackArrayDType)->arrayUnpackedElements();
+                // Handling the case where rhs is wider than lhs. StreamL does not require this
+                // since the combination of the left streaming operation and the implicit
+                // truncation in VL_ASSIGN_UNPACK automatically selects the left-most bits.
+                if (sWidth > dstBitWidth) {
+                    srcp
+                        = new AstSel{streamp->fileline(), srcp, sWidth - dstBitWidth, dstBitWidth};
+                }
+                srcp = new AstCvtPackedToUnpackArray{nodep->fileline(), srcp, dstp->dtypep()};
+            } else {
+                if (dWidth == 0) {
+                    srcp = new AstCvtPackedToDynArray{nodep->fileline(), srcp, dstp->dtypep()};
+                } else if (sWidth > dWidth) {
+                    srcp = new AstSel{streamp->fileline(), srcp, sWidth - dWidth, dWidth};
+                }
```

Comment:
What if sWidth <= dWidth?   Need an ASSERT_OBJ? Or does it work?

Reply:
Ah yes, that's the one last thing we haven't handle before this is good to merge. It should throw an error. I am planning to check this in an earlier stage, probably `V3Width`. So, maybe add an assert here, but throw the error in a previous stage?

Reply:
Makes sense.